packages/wonder-blocks-button/components/button.md

```diff
@@ -83,181 +131,325 @@ const styles = StyleSheet.create({
+    row: {
+        flexDirection: "row",
+        alignItems: "center",
+    },
+    button: {
+        marginRight: 10,
+    }
+});
+
+// NOTE: In actual code you would use BrowserRouter instead
+<MemoryRouter>
+    <View style={styles.row}>
+        <Button testId="button" href="/foo" clientNav={true} style={styles.button}>
+            Click me!
+        </Button>
+        <Switch>
+            <Route path="/foo">
+                <View id="foo">Hello, world!</View>
+            </Route>
+        </Switch>
+    </View>
+</MemoryRouter>
+```
+
+Buttons can have a `style` props which supports width, position, margin,
+and flex styles:
+
+TODO(kevinb): link to front matter section describing the `style` prop in
+more detail.
+
+### Best Practices
+
+In vertical layouts, buttons will stretch horizontally to fill the available
+space.  This is probably not what you want unless you're on a very narrow
+screen.
+
+```jsx
+const {View} = require("@khanacademy/wonder-blocks-core");
+
+<View>
+    <Button>
+        Label
+    </Button>
+</View>
+```
+
+This can be corrected by applying appropriate flex styles to the container.
+
+TODO(kevinb): link to front matter section describing why we use flexbox for
+everything.
+```jsx
+const {View} = require("@khanacademy/wonder-blocks-core");
+const {StyleSheet} = require("aphrodite");
+
+const styles = StyleSheet.create({
+    column: {
+        alignItems: "flex-start",
     },
-    sideMargins: {
+    row: {
+        flexDirection: "row",
+    },
+    gap: {
+        height: 16,
+    },
+    button: {
+        marginRight: 10,
+    },
+});
+
+<View>
+    <View style={styles.row}>
+        <Button>
+            Button in a row
+        </Button>
+    </View>
+    <View style={styles.gap} />
+    <View style={styles.column}>
+        <Button>
+            Button in a column
+        </Button>
+    </View>
+</View>
+```
+
+Layouts often specify a specific width of button.  When implementing such
+designs use `minWidth` instead of `width`.  `minWidth` allows the button
+to resize to fit the content whereas `width` does not.  This is important
+for international sites since sometimes strings for UI elements can be much
+longer in other languages.
+```jsx
+const {View} = require("@khanacademy/wonder-blocks-core");
+const {StyleSheet} = require("aphrodite");
+
+const styles = StyleSheet.create({
+    row: {
+        flexDirection: "row",
+    },
+    gap: {
+        height: 16,
+    },
+    goodButton: {
+        marginRight: 10,
+        minWidth: 144,
+    },
+    badButton: {
         marginRight: 10,
+        width: 144,
     },
 });
 
+<View>
+    <View style={styles.row}>
+        <Button
+            style={styles.goodButton}
+            kind="secondary"
+        >
+            label
+        </Button>
+        <Button
+            style={styles.goodButton}
+        >
+            label in a different language
+        </Button>
+    </View>
+    <View style={styles.gap} />
+    <View style={styles.row}>
```

Comment:
I think it'd be better to just remove the bad example from this as it makes it confusing as to which is the right one (or we should add text explaining which set of buttons are using which technique).